lib/identity_validations/service_provider_validation.rb

```diff
@@ -34,11 +34,11 @@ def redirect_uris_are_valid
       return if redirect_uris.blank?
 
       redirect_uris.each do |uri|
-        next if !uri.include?('*') && (uri_valid?(uri) || uri_custom_scheme_only?(uri))
-
-        errors.add(:redirect_uris, :invalid)
-        break
+        errors.add(:redirect_uris, "#{uri} contains invalid wildcards(*)") if uri.include?('*')
```

Comment:
This loops through all of the uris so that the Dashboard can show all the uris with errors instead of the first one the validator comes to. Also adds separate messages for wildcard violations vs parsing violations.